### ISSUE TITLE
Cannot read property '0' of null

### ISSUE THREAD

**Issue by rfshelley** [ci-dashboard/gitlab-ci-dashboard#7]
When loading the dashboard, I get `Cannot read property '0' of null`

Running against Gitlab 10.4.4 Enterprise. Gitlab is at `https://git.mydomain.io` (where mydomain is our internal hostname), and the whole path to the project is `https://git.mydomain.io/my-namespace/my-service`

Executed command: `gitlab-ci-dashboard --config ~/Projects/gitlabProjectsDashboard.json`
```
{
  "dashboard": {
    "config": {
      "gitlab": "git.mydomain.io",
      "token": "my_token",
      "apiVersion": 4
    },
    "projects": [
      {
        "description": "MyApp",
        "namespace": "my-namespace",
        "project": "my-service",
        "branch": "develop"
      }
    ]
  }
}
```
![image](https://user-images.githubusercontent.com/12369300/38711869-1cc5b5f4-3e7d-11e8-9e9e-b7924fc1e1db.png)

When checking the JS console, there are no errors reported. Looking at the Network tab, I can confirm that the project was retrieved:

**Request**
```
GET /api/v4/projects/my-namespace%2Fmy-service HTTP/1.1
Host: git.mydomain.io
Connection: keep-alive
Origin: http://localhost:8081
PRIVATE-TOKEN: my_token
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36
Accept: */*
DNT: 1
Referer: http://localhost:8081/?standalone=true
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
If-None-Match: W/"b84d8c457cfbea603a2488c5dfeebe3b"
```
**Response**
```
{"id":3374,"description":"","name":"my-service","name_with_namespace":"COP-FOH / my-service","path":"my-service","path_with_namespace":"my-namespace/my-service","created_at":"2014-11-05T17:09:46.000Z","default_branch":"develop","tag_list":[],"ssh_url_to_repo":"git@git.mydomain.io:my-namespace/my-service.git","http_url_to_repo":"https://git.mydomain.io/my-namespace/my-service.git","web_url":"https://git.mydomain.io/my-namespace/my-service","avatar_url":null,"star_count":1,"forks_count":5,"last_activity_at":"2018-04-13T00:39:27.119Z","_links":{"self":"http://git.mydomain.io/api/v4/projects/3374","merge_requests":"http://git.mydomain.io/api/v4/projects/3374/merge_requests","repo_branches":"http://git.mydomain.io/api/v4/projects/3374/repository/branches","labels":"http://git.mydomain.io/api/v4/projects/3374/labels","events":"http://git.mydomain.io/api/v4/projects/3374/events","members":"http://git.mydomain.io/api/v4/projects/3374/members"},"archived":false,"visibility":"public","resolve_outdated_diff_discussions":null,"container_registry_enabled":null,"issues_enabled":false,"merge_requests_enabled":true,"wiki_enabled":false,"jobs_enabled":true,"snippets_enabled":false,"shared_runners_enabled":true,"lfs_enabled":true,"creator_id":95,"namespace":{"id":145,"name":"COP-FOH","path":"my-namespace","kind":"group","full_path":"my-namespace","parent_id":null},"import_status":"none","import_error":null,"runners_token":"TDnKiMQkRuML_MmpEBdc","public_jobs":true,"ci_config_path":null,"shared_with_groups":[],"only_allow_merge_if_pipeline_succeeds":false,"request_access_enabled":true,"only_allow_merge_if_all_discussions_are_resolved":false,"printing_merge_request_link_enabled":true,"permissions":{"project_access":null,"group_access":{"access_level":50,"notification_level":3}}}
```
Was also able to retrieve the Branch details:
**Request**
```
GET /api/v4/projects/3374/repository/commits/develop HTTP/1.1
Host: git.mydomain.io
Connection: keep-alive
Origin: http://localhost:8081
PRIVATE-TOKEN: my_token
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36
Accept: */*
DNT: 1
Referer: http://localhost:8081/?standalone=true
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
If-None-Match: W/"f356f57b7dda58925838c47c3cace15d"
```
**Response**
```
{"id":"bb42b0ca5e41f90ac8733020e8709407da81a7f8","short_id":"bb42b0ca","title":"[ci skip] prepare for next development iteration","created_at":"2018-04-13T00:40:20.000+00:00","parent_ids":["abf03ffc1bb6040319c4d48e57b6d14c11b6f9ff"],"message":"[ci skip] prepare for next development iteration\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-04-13T00:40:20.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-04-13T00:40:20.000+00:00","stats":{"additions":5,"deletions":5,"total":10},"status":"skipped","last_pipeline":{"id":542175,"sha":"bb42b0ca5e41f90ac8733020e8709407da81a7f8","ref":"develop","status":"skipped"}}
```
Also ok getting tags:
**Request**
```
GET /api/v4/projects/3374/repository/tags HTTP/1.1
Host: git.mydomain.io
Connection: keep-alive
Origin: http://localhost:8081
PRIVATE-TOKEN: my_token
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36
Accept: */*
DNT: 1
Referer: http://localhost:8081/?standalone=true
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
If-None-Match: W/"9f360cd2fb6d0ff8d4b7926152187f69"
```
**Response**
```
[{"name":"myserviceparent-0.1.337","message":"[ci skip] copy for tag myserviceparent-0.1.337","commit":{"id":"abf03ffc1bb6040319c4d48e57b6d14c11b6f9ff","short_id":"abf03ffc","title":"[ci skip] prepare release myserviceparent-0.1.337","created_at":"2018-04-13T00:40:18.000+00:00","parent_ids":["a4e007723eb6bd94be0ba862fd6e5f31bbea278e"],"message":"[ci skip] prepare release myserviceparent-0.1.337\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-04-13T00:40:18.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-04-13T00:40:18.000+00:00"},"release":null},{"name":"myserviceparent-541448.a4e007723eb6bd94be0ba862fd6e5f31bbea278e","message":"[ci skip] prepare release myserviceparent-541448.a4e007723eb6bd94be0ba862fd6e5f31bbea278e","commit":{"id":"a4e007723eb6bd94be0ba862fd6e5f31bbea278e","short_id":"a4e00772","title":"JIRA-448, JIRA-454, JIRA-457: some message","created_at":"2018-04-12T11:55:49.000-07:00","parent_ids":["fffc940029f3c55eb3c653a5359daedc752ebf85"],"message":"JIRA-448, JIRA-454, JIRA-457: some message\n","author_name":"Foo Bar","author_email":"foo@mydomain.com","authored_date":"2018-04-12T11:55:49.000-07:00","committer_name":"Foo Bar","committer_email":"foo@mydomain.com","committed_date":"2018-04-12T11:55:49.000-07:00"},"release":null},{"name":"myserviceparent-0.1.336","message":"[ci skip] copy for tag myserviceparent-0.1.336","commit":{"id":"8f93151c68c17068b318830b571421777ce2fc73","short_id":"8f93151c","title":"[ci skip] prepare release myserviceparent-0.1.336","created_at":"2018-04-04T19:30:20.000+00:00","parent_ids":["a4b41319f682fb1615c48e42e49776868251a325"],"message":"[ci skip] prepare release myserviceparent-0.1.336\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-04-04T19:30:20.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-04-04T19:30:20.000+00:00"},"release":null},{"name":"myserviceparent-526508.a4b41319f682fb1615c48e42e49776868251a325","message":"[ci skip] prepare release myserviceparent-526508.a4b41319f682fb1615c48e42e49776868251a325","commit":{"id":"a4b41319f682fb1615c48e42e49776868251a325","short_id":"a4b41319","title":"Merge branch 'feature/JIRA-525' into 'develop'","created_at":"2018-04-04T18:41:43.000+00:00","parent_ids":["11d4ea0dffb6999308dc466c50800e4dd920a3d5","910f6015c8f7562be1aa8ae6b1c713ce0ad8bde4"],"message":"some message\n\nSee merge request my-namespace/myserviceservice!482","author_name":"Foo Bar","author_email":"foo@mydomain.com","authored_date":"2018-04-04T18:41:43.000+00:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-04-04T18:41:43.000+00:00"},"release":null},{"name":"myserviceparent-0.1.335","message":"[ci skip] copy for tag myserviceparent-0.1.335","commit":{"id":"b3d18baeaf847f772f313fac9c5cfbd2d1437c44","short_id":"b3d18bae","title":"[ci skip] prepare release myserviceparent-0.1.335","created_at":"2018-03-30T23:29:29.000+00:00","parent_ids":["c2c5deb034967ce68806dcb346e3cc78c6afb6e3"],"message":"[ci skip] prepare release myserviceparent-0.1.335\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-30T23:29:29.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-30T23:29:29.000+00:00"},"release":null},{"name":"myserviceparent-518836.c2c5deb034967ce68806dcb346e3cc78c6afb6e3","message":"[ci skip] prepare release myserviceparent-518836.c2c5deb034967ce68806dcb346e3cc78c6afb6e3","commit":{"id":"c2c5deb034967ce68806dcb346e3cc78c6afb6e3","short_id":"c2c5deb0","title":"Merge branch 'JIRA-497' into 'develop'","created_at":"2018-03-30T22:43:15.000+00:00","parent_ids":["df47fdf12e0533dd9658a17b282d20616b2c7db8","0acf76f76306a53d783cd1862c52cdb0bf1840d9"],"message":"Merge branch 'JIRA-497' into 'develop'\n\nJIRA-497\n\nSee merge request my-namespace/myserviceservice!481","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-30T22:43:15.000+00:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-30T22:43:15.000+00:00"},"release":null},{"name":"myserviceparent-0.1.334","message":"[ci skip] copy for tag myserviceparent-0.1.334","commit":{"id":"fcff36029fb78757c5a4f7859ecd97231bcf8bc1","short_id":"fcff3602","title":"[ci skip] prepare release myserviceparent-0.1.334","created_at":"2018-03-30T22:28:33.000+00:00","parent_ids":["fe4107cd74a9de756a6823543449b117b1ac55c5"],"message":"[ci skip] prepare release myserviceparent-0.1.334\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-30T22:28:33.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-30T22:28:33.000+00:00"},"release":null},{"name":"myserviceparent-518377.fe4107cd74a9de756a6823543449b117b1ac55c5","message":"[ci skip] prepare release myserviceparent-518377.fe4107cd74a9de756a6823543449b117b1ac55c5","commit":{"id":"fe4107cd74a9de756a6823543449b117b1ac55c5","short_id":"fe4107cd","title":"Merge branch 'feature/JIRA-432' into 'develop'","created_at":"2018-03-29T01:51:15.000+00:00","parent_ids":["a605b4d3077c5719bd6d2afaf7b18bb253e4f1d6","e5d4c8fabc17410d81a4658904636d4384c51070"],"message":"Merge branch 'feature/JIRA-432' into 'develop'\n\nJIRA-432: some message\n\nSee merge request my-namespace/myserviceservice!480","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-29T01:51:15.000+00:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-29T01:51:15.000+00:00"},"release":null},{"name":"myserviceparent-0.1.333","message":"[ci skip] copy for tag myserviceparent-0.1.333","commit":{"id":"6b168d6d838e4685d2ab8ff11050a10a04f82184","short_id":"6b168d6d","title":"[ci skip] prepare release myserviceparent-0.1.333","created_at":"2018-03-22T22:26:16.000+00:00","parent_ids":["3dd56516f20aae3daf17394caaffaa5b88f51b91"],"message":"[ci skip] prepare release myserviceparent-0.1.333\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-22T22:26:16.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-22T22:26:16.000+00:00"},"release":null},{"name":"myserviceparent-503924.3dd56516f20aae3daf17394caaffaa5b88f51b91","message":"[ci skip] prepare release myserviceparent-503924.3dd56516f20aae3daf17394caaffaa5b88f51b91","commit":{"id":"3dd56516f20aae3daf17394caaffaa5b88f51b91","short_id":"3dd56516","title":"JIRA-309/JIRA-412: some message","created_at":"2018-03-22T14:42:21.000-07:00","parent_ids":["8b015aef6c88ff1c4fd0430cd060b2211a1ac410"],"message":"JIRA-309/JIRA-412: some messge\n","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-22T14:42:21.000-07:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-22T14:42:21.000-07:00"},"release":null},{"name":"myserviceparent-0.1.332","message":"[ci skip] copy for tag myserviceparent-0.1.332","commit":{"id":"1b84fc45fcc36e916b125cdfee3a0a3ff3c2b9c3","short_id":"1b84fc45","title":"[ci skip] prepare release myserviceparent-0.1.332","created_at":"2018-03-22T02:44:08.000+00:00","parent_ids":["06b98d5c40ea141a2230ad800c521dcdac4e08e0"],"message":"[ci skip] prepare release myserviceparent-0.1.332\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-22T02:44:08.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-22T02:44:08.000+00:00"},"release":null},{"name":"myserviceparent-501953.06b98d5c40ea141a2230ad800c521dcdac4e08e0","message":"[ci skip] prepare release myserviceparent-501953.06b98d5c40ea141a2230ad800c521dcdac4e08e0","commit":{"id":"06b98d5c40ea141a2230ad800c521dcdac4e08e0","short_id":"06b98d5c","title":"Merge branch 'JIRA-412' into 'develop'","created_at":"2018-03-22T02:01:48.000+00:00","parent_ids":["ea4124074846534caaa3e7ed55b6e7302b2d7604","16f7e0dc6b2044f0356c35ce01033ea6ef7f6cb4"],"message":"Merge branch 'JIRA-412' into 'develop'\n\nJIRA-412: some message\n\nSee merge request my-namespace/myserviceservice!478","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-22T02:01:48.000+00:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-22T02:01:48.000+00:00"},"release":null},{"name":"myserviceparent-0.1.331","message":"[ci skip] copy for tag myserviceparent-0.1.331","commit":{"id":"6e9b222dddbcd2c27b7f03f38380397c6793876d","short_id":"6e9b222d","title":"[ci skip] prepare release myserviceparent-0.1.331","created_at":"2018-03-22T01:00:16.000+00:00","parent_ids":["ab0068e22dbdedabf4baadb6a600db0b0a582e6d"],"message":"[ci skip] prepare release myserviceparent-0.1.331\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-22T01:00:16.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-22T01:00:16.000+00:00"},"release":null},{"name":"myserviceparent-501848.ab0068e22dbdedabf4baadb6a600db0b0a582e6d","message":"[ci skip] prepare release myserviceparent-501848.ab0068e22dbdedabf4baadb6a600db0b0a582e6d","commit":{"id":"ab0068e22dbdedabf4baadb6a600db0b0a582e6d","short_id":"ab0068e2","title":"Merge branch 'foo' into 'develop'","created_at":"2018-03-22T00:12:57.000+00:00","parent_ids":["9c67a1d709650e26202f89aaf5760beedff96573","efefbc5db078f82f84fa3068dd1ea97492741b19"],"message":"Merge branch 'foo-bar' into 'develop'\n\nJIRA-521: some message\n\nSee merge request my-namespace/myserviceservice!476","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-22T00:12:57.000+00:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-22T00:12:57.000+00:00"},"release":null},{"name":"myserviceparent-0.1.330","message":"[ci skip] copy for tag myserviceparent-0.1.330","commit":{"id":"9836f2485deb86e1af2b72f83e06dc9347395a24","short_id":"9836f248","title":"[ci skip] prepare release myserviceparent-0.1.330","created_at":"2018-03-21T09:26:48.000+00:00","parent_ids":["08b3ec8ef6404c610dc991467601c8a2dd7d4438"],"message":"[ci skip] prepare release myserviceparent-0.1.330\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-21T09:26:48.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-21T09:26:48.000+00:00"},"release":null},{"name":"myserviceparent-499624.08b3ec8ef6404c610dc991467601c8a2dd7d4438","message":"[ci skip] prepare release myserviceparent-499624.08b3ec8ef6404c610dc991467601c8a2dd7d4438","commit":{"id":"08b3ec8ef6404c610dc991467601c8a2dd7d4438","short_id":"08b3ec8e","title":"Merge branch 'feature/JIRA-374' into 'develop'","created_at":"2018-03-21T08:41:22.000+00:00","parent_ids":["4e5f6fbea0070297342969dc45fea59c425424eb","69d792c16834a4b387f1ab096a690278c8e0af08"],"message":"Merge branch 'feature/JIRA-374' into 'develop'\n\nJIRA-374 some message\n\nSee merge request my-namespace/myserviceservice!477","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-21T08:41:22.000+00:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-21T08:41:22.000+00:00"},"release":null},{"name":"myserviceparent-0.1.329","message":"[ci skip] copy for tag myserviceparent-0.1.329","commit":{"id":"c175ef799522eac582104b989ce7464067ca6aac","short_id":"c175ef79","title":"[ci skip] prepare release myserviceparent-0.1.329","created_at":"2018-03-19T20:33:25.000+00:00","parent_ids":["26729fb26e0b5a500e54d1f077eca9d287072583"],"message":"[ci skip] prepare release myserviceparent-0.1.329\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-19T20:33:25.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-19T20:33:25.000+00:00"},"release":null},{"name":"myserviceparent-496128.26729fb26e0b5a500e54d1f077eca9d287072583","message":"[ci skip] prepare release myserviceparent-496128.26729fb26e0b5a500e54d1f077eca9d287072583","commit":{"id":"26729fb26e0b5a500e54d1f077eca9d287072583","short_id":"26729fb2","title":"Merge branch 'feature/JIRA-390' into 'develop'","created_at":"2018-03-19T19:46:53.000+00:00","parent_ids":["b300b2d8df62468560f3f09d0f7c2bfeb258e7a5","7372d0b248c6da184c4058efb332420686fa03b8"],"message":"Merge branch 'feature/JIRA-390' into 'develop'\n\nJIRA-390: some message\n\nSee merge request my-namespace/myserviceservice!475","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-19T19:46:53.000+00:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-19T19:46:53.000+00:00"},"release":null},{"name":"myserviceparent-0.1.328","message":"[ci skip] copy for tag myserviceparent-0.1.328","commit":{"id":"dff1ac34b02b18dd6812e8d10cb04e6d65a0a127","short_id":"dff1ac34","title":"[ci skip] prepare release myserviceparent-0.1.328","created_at":"2018-03-16T12:36:11.000+00:00","parent_ids":["8ba2a7119add2dfe4243b18b0ae8b296985b91b4"],"message":"[ci skip] prepare release myserviceparent-0.1.328\n","author_name":"gitlab-ci-runner","author_email":"nowhere@mydomain.com","authored_date":"2018-03-16T12:36:11.000+00:00","committer_name":"gitlab-ci-runner","committer_email":"nowhere@mydomain.com","committed_date":"2018-03-16T12:36:11.000+00:00"},"release":null},{"name":"myserviceparent-491578.8ba2a7119add2dfe4243b18b0ae8b296985b91b4","message":"[ci skip] prepare release myserviceparent-491578.8ba2a7119add2dfe4243b18b0ae8b296985b91b4","commit":{"id":"8ba2a7119add2dfe4243b18b0ae8b296985b91b4","short_id":"8ba2a711","title":"JIRA-343 some message","created_at":"2018-03-16T09:57:21.000+02:00","parent_ids":["d26ff1b509f4ee8c5b827b64d69d9d2e2e2b99d2"],"message":"JIRA-343 some message\n","author_name":"Foo Bar","author_email":"foo.bar@mydomain.com","authored_date":"2018-03-16T09:57:21.000+02:00","committer_name":"Foo Bar","committer_email":"foo.bar@mydomain.com","committed_date":"2018-03-16T09:57:21.000+02:00"},"release":null}]
```

**Comment by snowienator** [comment]
Any updates on this issue?

**Comment by iskren-dimov** [comment]
It seems that the issue is related to V4 API that you are using. I am getting the same problem, with version 3 it was working fine .
Of course, if you update GitLab to version 11, they have dropped V3 API :)

**Comment by WebconsultsEU** [comment]
Same problem here, no matter with api 3 or 4 on Gitlab  10.8.4

**Comment by EcaterinaGr** [comment]
Hello, 
So I had one repo that was showing up and another one that was displaying the error cannot read property of null.
Found out that this error is linked to the way the tags are managed. The first repo was working well because it did not have any tags. 

The problem is in this function:
export const getTopItemByName = (list) => {
  if (!Array.isArray(list) || list.length === 0) {
    return
  }
  return sort.desc(list).shift();
}

Replacing return sort.desc(list).shift(); with below block works well:
  sort.desc(list.map(function (tag) {
    return tag.name;
  }));

**Comment by loudstil** [comment]
Same issue here, @EcaterinaGr is right regarding the tags, the problem is that its trying to take the latest tag and it using semver-regex module that is trying to extract any format of version and when its not finding it it still trying to access item 0 on null : 

```javascript
return semvers.sort(function (v1, v2) {
    var sv1 = semverRegex().exec(v1)[0] || v1; // "0.2.0 m.alpha1.2105589" => 0.2.0
    var sv2 = semverRegex().exec(v2)[0] || v2; // TAG_PRE_V1B2_MERGE => null 
    return compare(sv1, sv2);
```

the solution I came up with is ugly but it works : 
```javascript
const default_version = '0.0.0';

return semvers.sort(function (v1, v2) {
    var v1_name = v1.name.replace(/[\-_]/g, " ");
    var sv1t = semverRegex().exec(v1_name);
    var sv1;
    if(sv1t != null) {
	sv1 = sv1t[0];
    } else {
        sv1 = default_version;
    }

    var v2_name = v2.name.replace(/[\-_]/g, " ");
    var sv2t = semverRegex().exec(v2_name);
    var sv2;
    if(sv2t != null) {
        sv2 = sv2t[0];
    } else {
        sv2 = default_version;
    }
    return compare(sv1, sv2);
});
```

**Comment by KITSSamBrown** [comment]
How do I implement this solution please? If I have to run another npm build, when I try to do this, it complains that config.build is undefined

**Comment by antoniomolram** [comment]
I can confirm that the this fix is needed for v4, I tested and now works!

**Comment by antoniomolram** [comment]
thank you @marek-ganko for your branch! ;)

**Comment by marek-ganko** [comment]
You are welcome @antoniomolram, now we need to wait for @emilianoeloi to build official npm package, hopefully he's still around :)

**Comment by marek-ganko** [comment]
In the meantime until my fix in PR #16 will be merged, you can use temporary published package with fixed for GitlabCI API v4: https://www.npmjs.com/package/gitlab-ci-v4-dashboard
